node/resolvers/search/index.ts

```diff
@@ -418,23 +460,50 @@ export const queries = {
 
   products: async (_: any, args: SearchArgs, ctx: Context) => {
     const {
-      clients: { search },
+      clients: { biggySearch },
+      vtex: { segment }
     } = ctx
-    const queryTerm = args.query
-    if (queryTerm == null || test(/[?&[\]=]/, queryTerm)) {
+    const { query, to, from, orderBy, simulationBehavior } = args
+
+    if (query == null || test(/[?&[\]=]/, query)) {
       throw new UserInputError(
-        `The query term contains invalid characters. query=${queryTerm}`
+        `The query term contains invalid characters. query=${query}`
       )
     }
```

Comment:
This arbitrary lock should be removed, our engine has no problem handling these characters 😄